gaugan/callbacks.py

```diff
@@ -31,32 +34,50 @@ def infer(self):
         )
         return self.model.predict([latent_vector, self.val_images[2]])
 
+    def log_to_tables(self, epoch, semantic_input, ground_truth, generated_images):
+        wandb_table = wandb.Table(columns=self.columns)
+        for i in range(4):
+            wandb_table.add_data(
+                wandb.Image(semantic_input[i]),
+                wandb.Image(ground_truth[i]),
+                wandb.Image(generated_images[i]),
+            )
+        wandb.log({f"GANMonitor Epoch {epoch}": wandb_table})
+
     def on_epoch_end(self, epoch, logs=None):
         if epoch == 0 or (epoch + 1) % self.epoch_interval == 0:
             generated_images = self.infer()
 
-            grid_row = min(generated_images.shape[0], 3)
-            fig, axarr = plt.subplots(4, 3, figsize=(18, grid_row * 6))
+            semantic_input = (self.val_images[0] + 1) / 2
```

Comment:
maybe this could be a function

Reply:
@scottire since it's just a one-liner preprocessing op, I think it's fine to leave it as it is.